server/transaction/TransactionService.java

```diff
@@ -62,11 +61,13 @@
     private final AtomicBoolean isOpen;
     private final AtomicBoolean commitRequested;
     private final AtomicInteger ongoingRequests;
+    private final int latencyMillis;
     private volatile String commitRequestID;
 
     public TransactionService(SessionService sessionSrv, TransactionStream stream, TransactionProto.Transaction.Open.Req request) {
         this.sessionSrv = sessionSrv;
         this.stream = stream;
+        this.latencyMillis = request.getLatencyMillis();
```

Comment:
Do you think it's worth renaming this to `networkLatencyMillis` for clarity, since we use the term _latency_ for multiple things? Or is it clear enough?

Reply:
I think `networkLatencyMillis` would be clearer here, you're right. But if we do that, we have to rename `latency_millis` to `network_latency_millis` in the protocol definitions too, right?

Reply:
Yeah I think that would be worth doing too.